instrumentation/google.golang.org/grpc/otelgrpc/interceptorinfo.go

```diff
@@ -0,0 +1,137 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelgrpc // import "go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+
+import (
+	"context"
+	"path"
+	"strings"
+
+	"google.golang.org/grpc"
+)
+
+// interceptorType is the flag to define which gRPC interceptor
+// the interceptorInfo object is.
+type interceptorType uint8
+
+const (
+	unaryClient interceptorType = iota
+	streamClient
+	unaryServer
+	streamServer
+)
+
+// interceptorInfo is the union of all arguments to four types of
+// gRPC interceptors, except for function types and function arguments
+// of invoker and streamer:
+// * invoker  grpc.UnaryInvoker (UnaryClient)
+// * streamer grpc.Streamer (StreamClient)
+// * stream   grpc.ServerStream (StreamServer)
+// * handler  grpc.UnaryHandler | grpc.StreamHandler (UnaryServer, StreamServer)
+// * req, reply, srv interface{} (UnaryClient, UnaryServer, StreamClient)
+// * cc       *grpc.ClientConn.
+type interceptorInfo struct {
+	ctx    context.Context
+	method string
+	desc   *grpc.StreamDesc
```

Comment:
Is this used anywhere?  I see a few places it's set, but I don't think it's accessed by any filters.

Reply:
Thanks for the comment! As you point out, it's not used in this PR! And first I left it so that others can utilize it later for other filters, `grpc.StreamDesc` is only used in the initialization to register stream, so we can remove this.